### ISSUE TITLE
--require-pragma option skips file when 'use strict'; is on first line

### ISSUE THREAD

**Issue by bguiz** [prettier/prettier#2962]
**Prettier 1.7.2**
[Playground link](https://prettier.io/playground/#%7B%22content%22%3A%22'use%20strict'%3B%5Cn%5Cn%2F**%5Cn%20*%20%40format%5Cn%20*%2F%5Cn%5Cnfunction%20foo(a%2Cb%2Cc)%20%7B%5Cn%20%20return%20%5C%22bar%5C%22%3B%5Cn%7D%5Cn%22%2C%22options%22%3A%7B%22ast%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22output2%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22printWidth%22%3A80%2C%22semi%22%3Atrue%2C%22singleQuote%22%3Atrue%2C%22requirePragma%22%3Atrue%2C%22tabWidth%22%3A2%2C%22trailingComma%22%3A%22all%22%2C%22useTabs%22%3Afalse%7D%7D)
```sh
--require-pragma
--single-quote
--trailing-comma all
```

**Input:**
```js
'use strict';

/**
 * @format
 */

function foo(a,b,c) {
  return "bar";
}
```

**Output:**
```jsx
'use strict';

/**
 * @format
 */

function foo(a,b,c) {
  return "bar";
}
```

**Expected behavior:**

```js
'use strict';

/**
 * @format
 */

function foo(a, b, c) {
  return 'bar';
}
```

### This works:

**Input:**
```js
/**
 * @format
 */

'use strict';

function foo(a,b,c) {
  return "bar";
}
```

**Output:**
```jsx
'use strict';

/**
 * @format
 */

function foo(a,b,c) {
  return "bar";
}
```

### Explanation

The bug appears to be in the implementation of the [`--require-pragma` option](https://github.com/prettier/prettier#--require-pragma).

> --require-pragma
> 
> Require a special comment, called a pragma, to be present in the file's first docblock comment in order for prettier to format it.
> 
> /**
>  * @prettier
>  */
> Valid pragmas are @prettier and @format.

It looks like `'use strict'` above the 1st doc-block throws this the `--require-pragma` option off.

**Comment by bguiz** [comment]
Note that the "playground link" above does not demonstrate the issue, as it does not support the `--require-pragma` option... perhaps it should?

**Comment by ikatyang** [comment]
Hi, thanks for the report. This is actually intended, where the first docblock comment means the docblock should be the **first** thing in the file, there shouldn't be anything before it except whitespaces.

For the `--require-pragma` option on playground, I've opened #2968 to track it.

**Comment by bguiz** [comment]
> Hi, thanks for the report. This is actually intended, where the first docblock comment means the docblock should be the first thing in the file, there shouldn't be anything before it except whitespaces.

The meaning of "the first docblock comment" does not imply that the docblock comment needs to be the first thing in the file, but rather that there are no other docblock comments that come before it.

I suggest that either:

1. The wording README is updated to reflect this position, or
2. Allow prettier to read the first docblock comment, as stated in the README

My preference is for the latter, as a developers (including myself) might prefer to do have `'use strict';` in the first line, or precede their first dockblock with `require` or `import` statements.

**Comment by bguiz** [comment]
My intent in raising this issue was not to ask a question, but rather to report a bug, so I'd suggest this label instead: https://github.com/prettier/prettier/labels/type%3Abug

**Comment by azz** [comment]
It's working as intended so not a bug. The problem with using the first docblock is

```js
function foo() {


```

**Comment by lydell** [comment]
@bguiz Feel free to send a PR improving the docs! :)

**Comment by bguiz** [comment]
@azz I would expect this pragma to flag in the following code:

```
function foo() {}

/**
 * @format
 * bar
 */
function bar() {}
```

I take your point that this looks weird, but disallowing this would also preclude things that *don't* look weird, such as the following:

```
'use strict';

/**
 * @format
 * @module MyModule does foo and bar
 */

function foo() {}

function bar() {}
```

tl;dr=Actually using "the first docblock comment" is more flexible
